test/tools/router_check/router.cc

```diff
@@ -183,7 +192,7 @@ bool RouterCheckTool::compareEntries(const std::string& expected_routes) {
     tool_config.route_ = config_->route(*tool_config.headers_, tool_config.random_value_);
 
     const std::string& test_name = check_config.test_name();
-    if (details_) {
+    if (details_ && !only_show_failures_) {
```

Comment:
i was wondering , if there are more things to print in the future out of the tests, would it be better if we put all the possible info in an object and then later iterate over the info and do overrides from the variables and print in just one place?

Reply:
I moved test results to a tests_ object, so any RouterCheckTool functions can edit/access that. The logic for determining what to output now lives at the end of compareEntries, as opposed to compareResults.